src/core/action.ts

```diff
@@ -28,6 +28,14 @@ export type IMiddleWareEvent = {
 let nextActionId = 1
 let currentActionContext: IMiddleWareEvent | null = null
 
+export function getSelf(): IStateTreeNode {
+    if (!currentActionContext || !currentActionContext.context)
```

Comment:
What about renaming currentActionContext to currentExecutionContext and add a field that tells if it is in view, action, and in the future, task execution?